example/boundary_exchange/boundary_exchange_driver.cpp

```diff
@@ -95,7 +78,96 @@ int main(int argc, char *argv[]) {
   forest_def.AddBC(edge_t({n[7], n[8]}));
 
   forest_def.AddInitialRefinement(parthenon::LogicalLocation(0, 1, 0, 0, 0));
-  pman.ParthenonInitPackagesAndMesh(forest_def);
+
+  return forest_def;
+}
+
+parthenon::forest::ForestDefinition n_blocks(int nblocks) {
+  using namespace parthenon::forest;
+  std::unordered_map<uint64_t, std::shared_ptr<Node>> nodes;
+  ForestDefinition forest_def;
+  parthenon::Real xoffset = 0.1;
+  parthenon::Real yoffset = 0.1;
+
+  for (int point = 0; point < 2 * nblocks; ++point) {
+    nodes[point] = Node::create(point, {-std::sin(point * M_PI / nblocks) + xoffset,
+                                        -std::cos(point * M_PI / nblocks) + yoffset});
+  }
+
+  using edge_t = parthenon::forest::Edge;
+  for (int point = 0; point < 2 * nblocks; ++point) {
+    forest_def.AddBC(
+        edge_t({nodes[point % (2 * nblocks)], nodes[(point + 1) % (2 * nblocks)]}));
+  }
```

Comment:
Are both nodes and edges required to define the forest? Or are you using nodes to define the edges?

Reply:
The forest is defined by a set of nodes, faces defined by an ordered set of four nodes, and boundary conditions on edges defined by a set of two nodes.